lib/p2p/Pool.ts

```diff
@@ -155,7 +155,7 @@ class Pool extends EventEmitter {
   private connectNode = async ({ addresses, nodePubKey }: NodeConnectionInfo) => {
     for (let n = 0; n < addresses.length; n += 1) {
       try {
-        await this.addOutbound(addresses[n], nodePubKey);
+        await this.addOutbound(addresses[n], nodePubKey, true);
```

Comment:
One more thought, this looks like we'd wait for the full timeout (currently a week, although I'm thinking we probably should shorten that) before we attempt the next address for a node that has multiple listening addresses. What do you think about making so we retry each address once before delaying and retrying? That would definitely require a bit of refactoring, but wanted to see what you thought.

Reply:
Sounds like a good idea, and shouldn't require serious refactoring, just to do one iteration without retries, and another one with.

@kilrau what do you say?

Reply:
Sounds like a good idea, and shouldn't require serious refactoring, just to do one iteration without retries, and another one with.

Reply:
Ok that should work, but then we'd basically only be retrying a single address from that point on right?

Reply:
Yes. I continued to discuss the option of doing the retries in parallel on the main comments thread.

Reply:
Answered in the main thread. In short: yes, retry logic PER listening address (otherwise this can take a very long time)